source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -592,6 +599,25 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+void ContextImpl::updateCertStats() {
+  // Update CA cert expiration time.
+  std::vector<Ssl::CertificateNameExpiration> ca_certs_expiration_time =
+      cert_validator_->getExpirationOfCerts();
+  for (const auto& ca_cert : ca_certs_expiration_time) {
+    cert_validator_->setExpirationOnCertStatsGivenCertName(ca_cert.name, ca_cert.expiration);
```

Comment:
I don't understand what's going on here. It looks like you're asking the validator for something, then giving that right back to the validator. Why not have a single call on the validator to have it do whatever needs to be done?

Reply:
Yea I agree. Maybe can just have a single function inside the validator. Maybe refreshCertStats() ?

Reply:
Replaced `getExpirationOfCerts ()` and `setExpirationOnCertStatsGivenCertName()` with `refreshCertStatsWithExpirationTime()`